### DISCUSSION TITLE
Sensor marked as failed after scheduler restart

### DISCUSSION THREAD

**Discussion by vlieven** [apache/airflow#33154] [General]
### Apache Airflow version

Other Airflow 2 version (please specify below)

### What happened

After a scheduler restart, a running sensor task was marked as failed.

As far as I can tell, the following sequence of events happened:
- a sensor task was running
- the scheduler was restarted
- after restart, the task was correctly adopted by the new scheduler
- the actual task state (success) did not match the expected task state (queued), causing the task to be marked as failed.

I believe this scenario is not adequately captured by the logic described here:
https://github.com/apache/airflow/blob/42465c5a9465fd77f3000117721e0ed1cc51c166/airflow/jobs/scheduler_job_runner.py#L748

This happened on Airflow 2.5.3

### What you think should happen instead

Relevant scheduler log:
```
{scheduler_job.py:687} ERROR - Executor reports task instance <TaskInstance: dag-name.task-name scheduled__2023-07-30T00:00:00+00:00 [queued]> finished (success) although the task says its queued. (Info: None) Was the task killed externally?""}"
```

This causes the following task log:
```
{taskinstance.py:2596} INFO - 0 downstream tasks scheduled from follow-on schedule check
{taskinstance.py:1080} INFO - Dependencies not met for <TaskInstance: dag-name.task-name  scheduled__2023-07-30T00:00:00+00:00 [failed]>, dependency 'Task Instance State' FAILED: Task is in the 'failed' state.
{local_task_job.py:151} INFO - Task is not able to be run
 ```
 
Given that the actual sensor state was `success`, it would be nicer to not mark it as `failed`, but rather `up_for_retry`.

### How to reproduce

I suppose you might be able to reproduce this if you get the timing exactly correct.

### Operating System

Debian 10

### Deployment

Other Docker-based deployment

### Deployment details

We're running this on kubernetes

### Are you willing to submit PR?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/apache/airflow/blob/main/CODE_OF_CONDUCT.md)

**Comment by RNHTTR** [comment]
Was the sensor a deferred sensor? If so, it [might have been fixed in 2.6.0](https://github.com/apache/airflow/pull/28586). This might be more appropriate for a discussion until it is reliably reproducible.

**Reply by vlieven** [reply]
No this was a regular sensor. It might be indeed more appropriate as a discussion, but I fear this will always be very tricky to reproduce.

**Reply by potiuk** [reply]
> No this was a regular sensor. It might be indeed more appropriate as a discussion, but I fear this will always be very tricky to reproduce.

True, but I'd say we have to have a reasonable confidence that this is an airflow issue even if we have no clear reproduction scenario. This is all about probability, not about 0/1 error /no error in this case. There is some probability that there is an Airflow issue. But currently the evidence we have does not support it. If we have no clear and simple reproduction scenario - then there is very close to 0 probability that someone will even attempt to reproduce it (how?) and get the probability of it being an Airflow issue higher. 

More likely than anything else (very close to 100%) is that if there is no more evidence coming, this issue will never get diagnosed and solved. So the best thing we can do is turn it into discussion hoping that a) original author will find more evidences b) somoene will stumble on it by searching and will provide more evidences of similar issue. 

This helps maintainers to work on our quest to only have real "issues" that can be dealt with in the issue list. It does not "kill it" straight away - having it as discussion that is open, gives a chance for someone to add extra evidences. It might be that we will not have more evidences in which case this one will die out as a discussion. Maybe the issue will reappear elsewhere with more evidences, new issue will be created, or maybe it was an intermittent problem. or maybe it's been already solved in one of the following releases? 

In all those cases keeping it as discussion is a good thing, keeping it as an open issue that most likely won't be solved without extra evidences is just noise that prevents from focusing on what's really important (and most important - what is much more actionable).

**Comment by potiuk** [comment]
Yeah. converting to discussion - we can always reopen if we find that the issue is still there.
